File: careercup/dynamicproblem/editdistance.go
package main

import (
	"fmt"
	"math"
)

func main()  {
	var str1 string = "sunday"
	var str2 string = "saturday"

	var m int = len(str1)
	var n int = len(str2)

	mat := make([][]int, m+1)
	for i := range mat {
		mat[i] = make([]int, n+1)
	}

	for i := 0; i <= m ; i++  {
		for j := 0; j <= n ;j++  {
			if i == 0 {
				mat[i][j] = j
			} else if j == 0 {
				mat[i][j] = i
			} else if str1[i-1] == str2[j-1] {
				mat[i][j] = mat[i-1][j-1]
			} else {
				mat[i][j] = int(math.Min(float64(mat[i-1][j-1]), math.Min( float64(mat[i][j-1]), float64(mat[i-1][j])))) + 1
			}
		}
	}

	fmt.Print(mat[m][n])
}

File: dynamicprogramming/src/EditDistance.java
/**
 * Created by chandresh.pancholi on 09/03/16.
 */
public class EditDistance {
    public static void main(String args[]) {
        String s1 = "sunday";
        String s2 = "saturday";

        int[][] temp = new int[s1.length() + 1][s2.length() + 1];

        for(int i = 0; i <= s1.length(); i++) {
            for(int j = 0; j <= s2.length(); j++) {
                if( i == 0){
                    temp[i][j] = j;
                } else if(j == 0) {
                    temp[i][j] = i;
                } else {
                    if(s1.charAt(i-1) == s2.charAt(j-1)) {
                        temp[i][j] = temp[i-1][j-1];
                    } else {
                        temp[i][j] = 1 + Math.min(temp[i-1][j], Math.min(temp[i][j-1], temp[i-1][j-1]));
                    }
                }
            }
        }

        System.out.print(temp[s1.length()][s2.length()]);
    }
}
